### ISSUE TITLE
[Bug]: Adding a custom mainnet chain is currently broken.

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-extension#13517]
![image (4)](https://user-images.githubusercontent.com/542863/152660566-49d523d0-060a-472a-ad1b-17516e79e82c.png)

I thought we had logged this already, but I see it's still happening. This is much worse with this morning's (albeit brief) Infura outage. Our users should always be able to self-host the services that make using their wallet possible.

**Comment by adonesky1** [comment]
> It’s a non-blocking warning so it actually doesn’t prevent any user from adding a custom RPC endpoint for Ethereum Mainnet.
I believe this is correct ☝️ 

As long as the RPC url returns `1` or `0x1` for `eth_chainId` the user should still be able to add the "alternative" mainnet provider.

**Comment by adonesky1** [comment]
Yeah that makes sense to me. Should be a few lines to do this since [this PR](https://github.com/MetaMask/metamask-extension/pull/12431) has merged
